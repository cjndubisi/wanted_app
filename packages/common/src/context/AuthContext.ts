import { Dispatch, Reducer, ReducerAction } from 'react';
import AsyncStorage from '@react-native-community/async-storage';
import { login, register, loginWithFacebookCredentials, loginWithGoogleCredentials } from '../api';
import { User, LoginResponse } from '../api/types';
import createDataContext from './createDataProvider';
import { LoginManager, AccessToken } from 'react-native-fbsdk';
import { GoogleSignin, statusCodes } from '@react-native-community/google-signin';

const AUTH_USER_TOKEN_KEY = 'AUTH_USER_TOKEN_KEY';

interface State {
  user: User;
  isLoading: boolean;
  auth_token: string;
  error: any;
  isSignedIn: boolean;
}

type Action = {
  type: AuthTypes;
  payload?: any;
};

enum AuthTypes {
  SIGNUP_SUCCESS = 'SIGNUP_SUCCESS',
  AUTH_SUCCESS = 'AUTH_SUCCESS',
  AUTH_FAILURE = 'AUTH_FAILURE',
  LOADING = 'LOADING',
}

type AuthReducer = Reducer<State, Action>;

const authReducer: AuthReducer = (prevState, action) => {
  switch (action.type) {
    case AuthTypes.AUTH_SUCCESS:
      return {
        ...prevState,
        user: action.payload.user,
        user_token: action.payload.user_token,
        isSignedIn: true,
        isLoading: false,
        error: undefined,
      };
    case AuthTypes.LOADING:
      return {
        ...prevState,
        isLoading: true,
        error: undefined,
      };
    case AuthTypes.AUTH_FAILURE:
      return {
        ...prevState,
        isLoading: false,
        error: action.payload,
      };
    default:
      return prevState;
  }
};

const loginAction = async (login: Promise<LoginResponse>, dispatch: (value: Action) => void) => {
  try {
    const { user, auth_token } = await login;
    console.log(user, auth_token);

    await AsyncStorage.setItem(AUTH_USER_TOKEN_KEY, auth_token);
    dispatch({
      type: AuthTypes.AUTH_SUCCESS,
      payload: {
        user,
        user_token: auth_token,
      },
    });
  } catch (error) {
    console.log(error);

    dispatch({
      type: AuthTypes.AUTH_FAILURE,
      payload: JSON.parse(error.message),
    });
  }
};

const authActions = (dispatch: Dispatch<ReducerAction<AuthReducer>>) => ({
  tryToLogin: async () => {
    GoogleSignin.configure();
    await AsyncStorage.clear();
    const auth_token = await AsyncStorage.getItem(AUTH_USER_TOKEN_KEY);
    if (auth_token) {
      dispatch({
        type: AuthTypes.AUTH_SUCCESS,
        payload: {
          user_token: auth_token,
        },
      });
    }
    return !!auth_token;
  },
  signUpWithEmail: async (info: Partial<User & { password: string }>) => {
    dispatch({ type: AuthTypes.LOADING });
    await loginAction(register(info), dispatch);
  },
  logInWithEmail: async (info: Partial<{ password: string; email: string }>) => {
    dispatch({ type: AuthTypes.LOADING });
    await loginAction(login(info), dispatch);
  },
  loginWithFacebook: async () => {
    const result = await LoginManager.logInWithPermissions(['public_profile', 'email']);
    if (result.isCancelled) return;
    let token = await AccessToken.getCurrentAccessToken();
    dispatch({ type: AuthTypes.LOADING });
    await loginAction(loginWithFacebookCredentials(token.accessToken), dispatch);
  },
  loginWithGoogle: async (token: string) => {
    try {
      await GoogleSignin.hasPlayServices();
      const { idToken } = await GoogleSignin.signIn();
      console.log(idToken);

      await loginAction(loginWithGoogleCredentials(idToken), dispatch);
    } catch (error) {
      if (
        [
          statusCodes.SIGN_IN_CANCELLED,
          statusCodes.IN_PROGRESS,
          statusCodes.PLAY_SERVICES_NOT_AVAILABLE,
        ].indexOf(error.code) !== -1
      ) {
        return;
      }

      throw error;
    }
  },
});

type DispatchAction = typeof authActions;

const INITIAL_STATE: State = {
  user: undefined,
  auth_token: undefined,
  isLoading: false,
  error: null,
  isSignedIn: false,
};

const { Provider, Context } = createDataContext<AuthReducer, DispatchAction, State>(
  authReducer,
  authActions,
  INITIAL_STATE
);

export { Provider as AuthProvider, Context as AuthContext };
