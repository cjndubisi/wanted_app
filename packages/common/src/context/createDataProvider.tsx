import React, { Dispatch, Reducer, ReducerAction, ReducerState, useReducer } from 'react';

type DispatchAction<T extends Reducer<any, any>, S> = (a: Dispatch<ReducerAction<T>>) => S;
var iscalled = false;

export default <
  T extends Reducer<S, any>,
  A extends DispatchAction<T, ReturnType<A> & { init?: () => Promise<void> }>,
  S
>(
  reducer: T,
  actions: A,
  defaultValue: ReducerState<T>
) => {
  type ContextType = ReturnType<A> & {
    state: ReducerState<T>;
  };

  const Context = React.createContext<ContextType>((undefined as any) as ContextType);

  const Provider = ({ children }: any) => {
    const [state, dispatch] = useReducer(reducer, defaultValue);
    const bindingActions = actions(dispatch);

    return (
      <Context.Provider
        value={{
          state,
          ...bindingActions,
        }}
      >
        {children}
      </Context.Provider>
    );
  };

  return {
    Context,
    Provider,
  };
};
