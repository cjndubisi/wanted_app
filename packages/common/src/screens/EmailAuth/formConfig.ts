export default [
  {
    placeholder: 'First Name',
    key: 'first_name',
    validation: (form): string => {
      const firstValid = form.first_name?.length > 2;
      return firstValid ? '' : 'First name is too short';
    },
  },
  {
    placeholder: 'Last Name',
    key: 'last_name',
    validation: (form): string => {
      const firstValid = form.last_name?.length > 2;
      return firstValid ? '' : 'Last name is too short';
    },
  },
  {
    placeholder: 'Email',
    key: 'email',
    validation: (form): string => {
      const emailValid = form.email?.match(/^([\w.%+-]+)@([\w-]+\.)+([\w]{2,})$/i);
      return emailValid ? '' : 'Invalid email';
    },
  },
  {
    placeholder: 'Password',
    key: 'password',
    validation: (form): string => {
      const passwordValid = form.password.match(
        /^(?=.*[a-z])(?=.*[A-Z])(?=.*[0-9])(?=.*[!@#\$%\^&\*])(?=.{6,})/
      );
      return passwordValid ? '' : 'Invalid Password format';
    },
  },
  {
    placeholder: 'Confirm Passwod',
    key: 'confirm_password',
    validation: (form): string => {
      if (form.password !== form.confirm_password) {
        return 'Password does not match';
      }
      return '';
    },
  },
];
