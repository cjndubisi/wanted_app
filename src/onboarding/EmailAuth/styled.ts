import styled from 'styled-components/native';

export const InputError = styled.Text<{ hasError: boolean }>`
  font-size: 12px;
  color: red;
  height: ${props => (props.hasError ? 13 : 0)}px;
  margin-bottom: 6px;
`;

export const Input = styled.TextInput`
  height: 40px;
  color: black;
  border-width: 1px;
  padding-left: 10px;
  padding-right: 10;
  font-size: 14px;
`;
