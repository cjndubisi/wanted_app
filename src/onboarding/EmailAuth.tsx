import React, { useState } from 'react';
import { TouchableOpacity } from 'react-native-gesture-handler';
import { StackNavigationProp } from '@react-navigation/stack';
import styled from 'styled-components/native';
import { Text, View } from 'react-native';

import { RootStackParamList } from '../router';
import { Container } from '../styles';
import { register } from '../api';

type SpashNavigationProp = StackNavigationProp<RootStackParamList, 'EmailAuth'>;
type Prop = {
  navigation: SpashNavigationProp;
};

const emailInfo = {
  first_name: '',
  last_name: '',
  email: '',
  password: '',
  confirm_password: '',
};
const InputError = styled.Text`
  font-size: 12px;
  color: red;
  height: 13px;
  margin-bottom: 6px;
`;

const Input = styled.TextInput`
  height: 40;
  color: black;
  border-width: 1;
  padding-left: 10;
  padding-right: 10;
  font-size: 14;
  margin-bottom: 10;
`;

const fieldsConfigurationOrder = [
  {
    placeholder: 'First Name',
    key: 'first_name',
    validation: (value): string => {
      const firstValid = value.length > 3;
      return firstValid ? '' : 'First name is too short';
    },
  },
  {
    placeholder: 'Last Name',
    key: 'last_name',
    validation: (value): string => {
      const firstValid = value.length > 3;
      return firstValid ? '' : 'Last name is too short';
    },
  },
  {
    placeholder: 'Email',
    key: 'email',
    validation: (value): string => {
      const emailValid = value.match(/^([\w.%+-]+)@([\w-]+\.)+([\w]{2,})$/i);
      return emailValid ? '' : 'Invalid email';
    },
  },
  {
    placeholder: 'Password',
    key: 'password',
    validation: (value): string => {
      const passwordValid = value.length >= 6;
      return passwordValid ? '' : 'Password is too short';
    },
  },
  {
    placeholder: 'Confirm Password',
    key: 'confirm_password',
    validation: (value): string => {
      const passwordValid = value.length > 6;
      return passwordValid ? '' : 'Password does not match';
    },
  },
];

type ButtonProps = {
  title: string;
  onPress(): void;
};

const Button = ({ title, onPress }: ButtonProps) => (
  <TouchableOpacity activeOpacity={0.8} onPress={onPress}>
    <Text>{title}</Text>
  </TouchableOpacity>
);

export default () => {
  const [info, setInfo] = useState<typeof emailInfo>(emailInfo);
  const [formError, setFormError] = useState<typeof emailInfo | {}>(emailInfo);
  const signUp = async () => {
    try {
      const request = info;
      delete request.confirm_password;
      const response = await register(request);
      console.log(response);
    } catch (error) {
      console.log(JSON.stringify(error));
    }
  };

  const validate = () => {
    const errors: typeof emailInfo | {} = fieldsConfigurationOrder.reduce((acc, next) => {
      const error = next.validation(info[next.key]);
      if (error !== '') {
        acc[next.key] = next.validation(info[next.key]);
      }
      return acc;
    }, {});

    if (Object.keys(errors).length > 0) {
      return setFormError(errors);
    }

    signUp();
  };

  const inputs = fieldsConfigurationOrder.map(input => (
    <View key={`input_${input.key}`}>
      <InputError key={`error_${input.key}`}>{formError[input.key]}</InputError>
      <Input
        key={input.key}
        placeholder={input.placeholder}
        secureTextEntry={input.key.indexOf('password') !== -1}
        onChangeText={(text: string) => setInfo({ ...info, [input.key]: text })}
        value={info[input.key]}
      />
    </View>
  ));

  return (
    <Container>
      {inputs}
      <Button title="Sign Up with Email" onPress={validate} />
    </Container>
  );
};
