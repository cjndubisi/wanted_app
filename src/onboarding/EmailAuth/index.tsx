import { StackNavigationProp } from '@react-navigation/stack';
import React, { useState } from 'react';
import { View } from 'react-native';
import { register } from '../../api/authentication';
import { User } from '../../api/types';
import { RootStackParamList } from '../../router';
import Button from '../../shared/components/Button';
import { Container, Text } from '../../shared/styled';
import formConfig from './formConfig';
import { Input, InputError } from './styled';

type SpashNavigationProp = StackNavigationProp<RootStackParamList, 'EmailAuth'>;
type Prop = {
  navigation: SpashNavigationProp;
};

type FormState = Partial<User & { password: string, confirm_password: string }>;
type FormErrorState = FormState;

export default () => {
  const [info, setInfo] = useState<FormState>({});
  const [formError, setFormError] = useState<FormErrorState>({});
  const signUp = async () => {
    try {
      const request = info;
      delete request.confirm_password;
      const response = await register(request);
    } catch (error) { }
  };

  const validate = () => {
    const errors: FormErrorState = formConfig.reduce((acc, next) => {
      acc[next.key] = next.validation(info[next.key] ?? '');
      return acc;
    }, {});

    if (Object.keys(errors).length > 0) {
      return setFormError(errors);
    }

    signUp();
  };

  return (
    <Container>
      {formConfig.map(input => (
        <View key={`input_${input.key}`}>
          <Text>{input.placeholder}</Text>
          <Input
            key={input.key}
            placeholder={input.placeholder}
            secureTextEntry={input.key.indexOf('password') !== -1}
            onChangeText={(text: string) => {
              setInfo({ ...info, [input.key]: text })
              setFormError({ ...formError, [input.key]: '' })
            }}
            value={info[input.key]}
          />
          <InputError key={`error_${input.key}`} hasError={!!formError[input.key]?.length}>
            {formError[input.key]}
          </InputError>
        </View>)
      )}
      <Button bold
        title='Sign Up with Email'
        onPress={validate}
        titleColor='white'
        backgroundColor='brown' />
    </Container>
  );
};
