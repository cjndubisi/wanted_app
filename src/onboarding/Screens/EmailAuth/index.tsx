import { StackNavigationProp } from '@react-navigation/stack';
import React, { useState } from 'react';
import { Platform, View } from 'react-native';
import { RootStackParamList } from '../../../router';
import { User } from '../../../shared/api/types';
import Button from '../../../shared/components/Button';
import { AuthContext } from '../../../shared/context/AuthContext';
import { Container, H1, InputCaption, Label, Text } from '../../../shared/styled';
import formConfig from './formConfig';
import { Input, InputError } from './styled';

type SpashNavigationProp = StackNavigationProp<RootStackParamList, 'EmailAuth'>;
type FormState = Partial<User & { password: string; confirm_password: string }>;
type FormErrorState = FormState;

export default ({ navigation }: { navigation: SpashNavigationProp }) => {
  navigation.setOptions({
    headerShown: Platform.OS != 'web',
  });

  const { signUpWithEmail, state } = React.useContext(AuthContext);
  const [info, setInfo] = useState<FormState>({});
  const [formError, setFormError] = useState<FormErrorState>({});
  // To prevent rendering page, as cannot dispatch { error: null } to contect from here
  const [showingAPIError, setShowingAPIError] = useState<boolean>(false);

  const signUp = async () => {
    const errors: FormErrorState = formConfig.reduce((acc, next) => {
      const error = next.validation(info[next.key] ?? '');
      // update empty object only if error exist
      if (error !== '') {
        acc[next.key] = error;
      }
      return acc;
    }, {});

    if (Object.keys(errors).length > 0) {
      return setFormError(errors);
    }

    // sign up
    let request = { ...info };
    delete request.confirm_password;

    setShowingAPIError(false);
    await signUpWithEmail(request);
  };

  if (state.error && !showingAPIError) {
    const error = state.error;
    let viewError = {};
    Object.keys(error).forEach((key) => {
      // does the key match any form field
      if (info[key] !== undefined) {
        viewError[key] = error[key][0];
      }
    });

    if (Object.keys(viewError).length !== 0) {
      setFormError({ ...formError, ...viewError });
    }
    setShowingAPIError(true);
  }

  return (
    <Container style={{ marginTop: 44 }}>
      <View>
        <View style={{ marginBottom: 20, marginTop: 20 }}>
          <H1>Wanted</H1>
          <Text>Create an account buy and sell services, product, jobs and more.</Text>
        </View>

        {formConfig.map((input) => (
          <View style={{ marginBottom: 12 }} key={`input_${input.key}`}>
            <View
              style={{
                flex: 1,
                display: 'flex',
                justifyContent: 'space-between',
                flexDirection: 'row',
              }}
            >
              <Label bold>{input.placeholder}</Label>
              <InputError key={`error_${input.key}`} hasError={!!formError[input.key]?.length}>
                {formError[input.key]}
              </InputError>
            </View>
            <Input
              key={input.key}
              secureTextEntry={input.key.indexOf('password') !== -1}
              onChangeText={(text: string) => {
                setInfo({ ...info, [input.key]: text });
                setFormError({ ...formError, [input.key]: '' });
              }}
              value={info[input.key] || ''}
            />
            {input.key === 'password' ? (
              <InputCaption>
                {'Requires at least an Uppercase letter, a symbol and a number'}
              </InputCaption>
            ) : null}
          </View>
        ))}
        <Button
          bold
          title="Sign up with email"
          onPress={signUp}
          titleColor="white"
          backgroundColor="brown"
        />
      </View>
    </Container>
  );
};
