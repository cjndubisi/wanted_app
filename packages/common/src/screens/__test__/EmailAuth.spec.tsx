import { BaseNavigationContainer } from '@react-navigation/core';
import { createStackNavigator } from '@react-navigation/stack';
import { fireEvent, render, ReactTestInstanceExtended } from '@testing-library/react-native';
import React from 'react';
import { create } from 'react-test-renderer';
import EmailAuth, { FormState } from '../EmailAuth';
import { AuthProvider } from '../../context/AuthContext';

const withNavigation = ({
  screens = {},
}: {
  screens: { [component: string]: { component: any } };
}) => {
  return class extends React.Component {
    render() {
      const Stack = createStackNavigator();
      return (
        <BaseNavigationContainer>
          <Stack.Navigator>
            {Object.keys(screens).map((name) => (
              <Stack.Screen key={name} name={name} component={screens[name].component} />
            ))}
          </Stack.Navigator>
        </BaseNavigationContainer>
      );
    }
  };
};

const renderWithNavigation = ({ screens = {} }) => {
  const NavigationComponent = withNavigation({ screens });
  const Component = withProviders(NavigationComponent);
  return { ...render(<Component />) };
};

const withProviders = (Component: any) => {
  const signUpWithEmail = jest.fn();

  return class extends React.Component {
    render() {
      return (
        <AuthProvider value={{ signUpWithEmail }}>
          <Component />
        </AuthProvider>
      );
    }
  };
};

it('renders correctly', () => {
  const EmailNavigation = withNavigation({ screens: { EmailAuth: { component: EmailAuth } } });
  const Auth = withProviders(EmailNavigation);
  const tree = create(<Auth />).toJSON();

  expect(tree).toMatchSnapshot();
});

test('cannot submit with empty fields', async () => {
  const screens = { screens: { EmailAuth: { component: EmailAuth } } };
  const { findByLabelText, getByTitle, getByText } = renderWithNavigation(screens);

  fireEvent.press(getByText(/Sign up with email/i));

  await expect(findByLabelText('Last name is too short')).toBeTruthy();
});

test('test can sign up', async () => {
  const formInput: FormState = {
    first_name: 'random',
    last_name: 'random',
    email: 'random@random.com',
    password: 'random',
    confirm_password: 'random',
  };
  const screens = { screens: { EmailAuth: { component: EmailAuth } } };
  const { getByLabelText } = renderWithNavigation(screens);
  const keys: { [T in keyof typeof formInput] } = Object.keys(formInput).reduce((acc, next) => {
    acc[next] = next;
    return acc;
  }, {});
  const elements = Object.keys(formInput).reduce((acc, name) => {
    acc[name] = getByLabelText(name);
    return acc;
  }, {});
  Object.keys(elements).forEach((item) => {
    fireEvent.changeText(elements[keys.email], formInput[item]);
  });

  // expect(getByTestId('title').props.children).toMatch('Home page');
  // fireEvent.press(getByText(/About page/i));
  // await expect(findByText('About page')).toBeTruthy();
});
