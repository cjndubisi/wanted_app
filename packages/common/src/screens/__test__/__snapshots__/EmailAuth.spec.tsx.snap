// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RNCSafeAreaView
    onInsetsChange={[Function]}
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      onLayout={[Function]}
      style={
        Object {
          "flex": 1,
          "overflow": "hidden",
        }
      }
    >
      <View
        pointerEvents="box-none"
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <View
          accessibilityElementsHidden={false}
          closing={false}
          gestureVelocityImpact={0.3}
          importantForAccessibility="auto"
          onClose={[Function]}
          onGestureBegin={[Function]}
          onGestureCanceled={[Function]}
          onOpen={[Function]}
          onTransitionStart={[Function]}
          pointerEvents="box-none"
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
          transitionSpec={
            Object {
              "close": Object {
                "animation": "spring",
                "config": Object {
                  "damping": 500,
                  "mass": 3,
                  "overshootClamping": true,
                  "restDisplacementThreshold": 0.01,
                  "restSpeedThreshold": 0.01,
                  "stiffness": 1000,
                },
              },
              "open": Object {
                "animation": "spring",
                "config": Object {
                  "damping": 500,
                  "mass": 3,
                  "overshootClamping": true,
                  "restDisplacementThreshold": 0.01,
                  "restSpeedThreshold": 0.01,
                  "stiffness": 1000,
                },
              },
            }
          }
        >
          <View
            pointerEvents="box-none"
            style={
              Object {
                "flex": 1,
                "marginTop": 64,
              }
            }
          >
            <View
              collapsable={false}
              onGestureHandlerEvent={[Function]}
              onGestureHandlerStateChange={[Function]}
              style={
                Object {
                  "flex": 1,
                  "transform": Array [
                    Object {
                      "translateX": 0,
                    },
                    Object {
                      "translateX": 0,
                    },
                  ],
                }
              }
            >
              <View
                pointerEvents="none"
                style={
                  Object {
                    "backgroundColor": "#fff",
                    "bottom": 0,
                    "left": 0,
                    "position": "absolute",
                    "shadowColor": "#000",
                    "shadowOffset": Object {
                      "height": 1,
                      "width": -1,
                    },
                    "shadowOpacity": 0.3,
                    "shadowRadius": 5,
                    "top": 0,
                    "width": 3,
                  }
                }
              />
              <View
                style={
                  Array [
                    Object {
                      "flex": 1,
                      "overflow": "hidden",
                    },
                    Array [
                      Object {
                        "backgroundColor": "rgb(242, 242, 242)",
                      },
                      undefined,
                    ],
                  ]
                }
              >
                <View
                  style={
                    Object {
                      "flex": 1,
                      "flexDirection": "column-reverse",
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "flex": 1,
                      }
                    }
                  >
                    <View
                      style={
                        Array [
                          Object {
                            "alignSelf": "center",
                            "display": "flex",
                            "flexBasis": 0,
                            "flexDirection": "column",
                            "flexGrow": 1,
                            "flexShrink": 1,
                            "justifyContent": "space-between",
                            "maxWidth": 467,
                          },
                        ]
                      }
                    >
                      <View
                        style={
                          Object {
                            "alignItems": "center",
                            "backgroundColor": "rgba(52, 52, 52, 0.3)",
                            "display": "none",
                            "flex": 1,
                            "flexDirection": "column",
                            "height": 1334,
                            "justifyContent": "center",
                            "left": 0,
                            "position": "absolute",
                            "top": 0,
                            "width": 750,
                            "zIndex": 1,
                          }
                        }
                      >
                        <View
                          style={
                            Object {
                              "alignItems": "center",
                              "backgroundColor": "white",
                              "borderRadius": 10,
                              "height": 100,
                              "justifyContent": "center",
                              "width": 100,
                            }
                          }
                        >
                          <ActivityIndicator
                            animating={false}
                            color="gray"
                            hidesWhenStopped={true}
                            size="large"
                          />
                        </View>
                      </View>
                      <View
                        style={
                          Object {
                            "margin": 20,
                            "marginTop": 44,
                          }
                        }
                      >
                        <View
                          style={
                            Object {
                              "marginBottom": 20,
                              "marginTop": 20,
                            }
                          }
                        >
                          <Text
                            style={
                              Array [
                                Object {
                                  "fontSize": 24,
                                  "fontWeight": "bold",
                                  "marginBottom": 10,
                                },
                              ]
                            }
                          >
                            Wanted
                          </Text>
                          <Text
                            style={
                              Array [
                                Object {},
                                Object {
                                  "color": "black",
                                  "marginBottom": 4,
                                },
                              ]
                            }
                          >
                            Create an account buy and sell services, product, jobs and more.
                          </Text>
                        </View>
                        <Text
                          style={
                            Array [
                              Object {
                                "color": "red",
                                "fontSize": 12,
                                "height": 13,
                                "marginBottom": 6,
                              },
                            ]
                          }
                        />
                        <View
                          style={
                            Object {
                              "marginBottom": 12,
                            }
                          }
                        >
                          <View
                            style={
                              Object {
                                "display": "flex",
                                "flex": 1,
                                "flexDirection": "row",
                                "justifyContent": "space-between",
                              }
                            }
                          >
                            <Text
                              style={
                                Array [
                                  Object {},
                                  Object {
                                    "color": "black",
                                    "fontSize": 12,
                                    "marginBottom": 4,
                                  },
                                ]
                              }
                            >
                              First Name
                            </Text>
                            <Text
                              accessibilityLabel="error_first_name"
                              style={
                                Array [
                                  Object {
                                    "color": "red",
                                    "fontSize": 12,
                                    "height": 13,
                                    "marginBottom": 6,
                                  },
                                ]
                              }
                            />
                          </View>
                          <TextInput
                            accessibilityLabel="first_name"
                            allowFontScaling={true}
                            onChangeText={[Function]}
                            rejectResponderTermination={true}
                            secureTextEntry={false}
                            style={
                              Array [
                                Object {
                                  "backgroundColor": "white",
                                  "borderRadius": 5,
                                  "color": "black",
                                  "fontSize": 14,
                                  "height": 40,
                                  "paddingLeft": 10,
                                  "paddingRight": 10,
                                },
                              ]
                            }
                            underlineColorAndroid="transparent"
                            value=""
                          />
                        </View>
                        <View
                          style={
                            Object {
                              "marginBottom": 12,
                            }
                          }
                        >
                          <View
                            style={
                              Object {
                                "display": "flex",
                                "flex": 1,
                                "flexDirection": "row",
                                "justifyContent": "space-between",
                              }
                            }
                          >
                            <Text
                              style={
                                Array [
                                  Object {},
                                  Object {
                                    "color": "black",
                                    "fontSize": 12,
                                    "marginBottom": 4,
                                  },
                                ]
                              }
                            >
                              Last Name
                            </Text>
                            <Text
                              accessibilityLabel="error_last_name"
                              style={
                                Array [
                                  Object {
                                    "color": "red",
                                    "fontSize": 12,
                                    "height": 13,
                                    "marginBottom": 6,
                                  },
                                ]
                              }
                            />
                          </View>
                          <TextInput
                            accessibilityLabel="last_name"
                            allowFontScaling={true}
                            onChangeText={[Function]}
                            rejectResponderTermination={true}
                            secureTextEntry={false}
                            style={
                              Array [
                                Object {
                                  "backgroundColor": "white",
                                  "borderRadius": 5,
                                  "color": "black",
                                  "fontSize": 14,
                                  "height": 40,
                                  "paddingLeft": 10,
                                  "paddingRight": 10,
                                },
                              ]
                            }
                            underlineColorAndroid="transparent"
                            value=""
                          />
                        </View>
                        <View
                          style={
                            Object {
                              "marginBottom": 12,
                            }
                          }
                        >
                          <View
                            style={
                              Object {
                                "display": "flex",
                                "flex": 1,
                                "flexDirection": "row",
                                "justifyContent": "space-between",
                              }
                            }
                          >
                            <Text
                              style={
                                Array [
                                  Object {},
                                  Object {
                                    "color": "black",
                                    "fontSize": 12,
                                    "marginBottom": 4,
                                  },
                                ]
                              }
                            >
                              Email
                            </Text>
                            <Text
                              accessibilityLabel="error_email"
                              style={
                                Array [
                                  Object {
                                    "color": "red",
                                    "fontSize": 12,
                                    "height": 13,
                                    "marginBottom": 6,
                                  },
                                ]
                              }
                            />
                          </View>
                          <TextInput
                            accessibilityLabel="email"
                            allowFontScaling={true}
                            onChangeText={[Function]}
                            rejectResponderTermination={true}
                            secureTextEntry={false}
                            style={
                              Array [
                                Object {
                                  "backgroundColor": "white",
                                  "borderRadius": 5,
                                  "color": "black",
                                  "fontSize": 14,
                                  "height": 40,
                                  "paddingLeft": 10,
                                  "paddingRight": 10,
                                },
                              ]
                            }
                            underlineColorAndroid="transparent"
                            value=""
                          />
                        </View>
                        <View
                          style={
                            Object {
                              "marginBottom": 12,
                            }
                          }
                        >
                          <View
                            style={
                              Object {
                                "display": "flex",
                                "flex": 1,
                                "flexDirection": "row",
                                "justifyContent": "space-between",
                              }
                            }
                          >
                            <Text
                              style={
                                Array [
                                  Object {},
                                  Object {
                                    "color": "black",
                                    "fontSize": 12,
                                    "marginBottom": 4,
                                  },
                                ]
                              }
                            >
                              Password
                            </Text>
                            <Text
                              accessibilityLabel="error_password"
                              style={
                                Array [
                                  Object {
                                    "color": "red",
                                    "fontSize": 12,
                                    "height": 13,
                                    "marginBottom": 6,
                                  },
                                ]
                              }
                            />
                          </View>
                          <TextInput
                            accessibilityLabel="password"
                            allowFontScaling={true}
                            onChangeText={[Function]}
                            rejectResponderTermination={true}
                            secureTextEntry={true}
                            style={
                              Array [
                                Object {
                                  "backgroundColor": "white",
                                  "borderRadius": 5,
                                  "color": "black",
                                  "fontSize": 14,
                                  "height": 40,
                                  "paddingLeft": 10,
                                  "paddingRight": 10,
                                },
                              ]
                            }
                            underlineColorAndroid="transparent"
                            value=""
                          />
                          <Text
                            style={
                              Array [
                                Object {},
                                Object {
                                  "color": "gray",
                                  "fontSize": 10,
                                  "marginBottom": 4,
                                  "marginTop": 4,
                                },
                              ]
                            }
                          >
                            Requires at least an Uppercase letter, a symbol and a number
                          </Text>
                        </View>
                        <View
                          style={
                            Object {
                              "marginBottom": 12,
                            }
                          }
                        >
                          <View
                            style={
                              Object {
                                "display": "flex",
                                "flex": 1,
                                "flexDirection": "row",
                                "justifyContent": "space-between",
                              }
                            }
                          >
                            <Text
                              style={
                                Array [
                                  Object {},
                                  Object {
                                    "color": "black",
                                    "fontSize": 12,
                                    "marginBottom": 4,
                                  },
                                ]
                              }
                            >
                              Confirm Passwod
                            </Text>
                            <Text
                              accessibilityLabel="error_confirm_password"
                              style={
                                Array [
                                  Object {
                                    "color": "red",
                                    "fontSize": 12,
                                    "height": 13,
                                    "marginBottom": 6,
                                  },
                                ]
                              }
                            />
                          </View>
                          <TextInput
                            accessibilityLabel="confirm_password"
                            allowFontScaling={true}
                            onChangeText={[Function]}
                            rejectResponderTermination={true}
                            secureTextEntry={true}
                            style={
                              Array [
                                Object {
                                  "backgroundColor": "white",
                                  "borderRadius": 5,
                                  "color": "black",
                                  "fontSize": 14,
                                  "height": 40,
                                  "paddingLeft": 10,
                                  "paddingRight": 10,
                                },
                              ]
                            }
                            underlineColorAndroid="transparent"
                            value=""
                          />
                        </View>
                        <View
                          accessible={true}
                          focusable={true}
                          isTVSelectable={true}
                          onClick={[Function]}
                          onResponderGrant={[Function]}
                          onResponderMove={[Function]}
                          onResponderRelease={[Function]}
                          onResponderTerminate={[Function]}
                          onResponderTerminationRequest={[Function]}
                          onStartShouldSetResponder={[Function]}
                          style={
                            Object {
                              "alignItems": "center",
                              "backgroundColor": "brown",
                              "borderRadius": 5,
                              "height": 40,
                              "justifyContent": "center",
                              "marginTop": 10,
                              "opacity": 1,
                            }
                          }
                        >
                          <Text
                            style={
                              Array [
                                Object {},
                                Object {
                                  "color": "black",
                                  "marginBottom": 4,
                                },
                                Object {
                                  "color": "white",
                                },
                              ]
                            }
                          >
                            Sign up with email
                          </Text>
                        </View>
                      </View>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
    <View
      pointerEvents="box-none"
      style={
        Object {
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    >
      <View
        accessibilityElementsHidden={false}
        importantForAccessibility="auto"
        onLayout={[Function]}
        pointerEvents="box-none"
        style={null}
      >
        <View
          pointerEvents="none"
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "opacity": 1,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "rgb(255, 255, 255)",
                "borderBottomColor": "rgb(224, 224, 224)",
                "flex": 1,
                "shadowColor": "rgb(224, 224, 224)",
                "shadowOffset": Object {
                  "height": 0.5,
                  "width": 0,
                },
                "shadowOpacity": 0.85,
                "shadowRadius": 0,
              }
            }
          />
        </View>
        <View
          pointerEvents="box-none"
          style={
            Object {
              "height": 64,
              "maxHeight": undefined,
              "minHeight": undefined,
              "opacity": undefined,
              "transform": undefined,
            }
          }
        >
          <View
            pointerEvents="none"
            style={
              Object {
                "height": 20,
              }
            }
          />
          <View
            pointerEvents="box-none"
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "flexDirection": "row",
                "justifyContent": "center",
              }
            }
          >
            <View
              pointerEvents="box-none"
              style={
                Object {
                  "marginHorizontal": 18,
                  "opacity": 1,
                }
              }
            >
              <Text
                accessibilityRole="header"
                numberOfLines={1}
                onLayout={[Function]}
                style={
                  Object {
                    "color": "rgb(28, 28, 30)",
                    "fontSize": 17,
                    "fontWeight": "600",
                  }
                }
              >
                EmailAuth
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RNCSafeAreaView>
</View>
`;
